microprofile-health/README.adoc

```diff
@@ -0,0 +1,425 @@
+is detailed in the MicroProfile Health specification. Generally, the liveness
+procedures determine whether the application should be restarted while readiness
+procedures determine whether it makes sense to contact the application with requests.
+
+
+== Negative health check procedures
+
+In this section, we extend our `Database connection health check` with the option of
+stating that our application is not ready to process requests as the underlying
+database connection cannot be established. For simplicity reasons, we only determine
+whether the database is accessible or not by a configuration property.
+
+To use MicroProfile Config configuration values we first need to add the Config API
+dependency to our application:
+
+[source,xml]
+----
+<!-- Import the MicroProfile Config API, we use provided scope as the API is included in WildFly -->
+<dependency>
+  <groupId>org.eclipse.microprofile.config</groupId>
+  <artifactId>microprofile-config-api</artifactId>
+  <version>1.3</version>
+  <scope>provided</scope>
+</dependency>
+----
+
+Update the `org.quickstart.microprofile.health.DatabaseConnectionHealthCheck` class:
+
+[source,java]
+----
+package org.quickstart.microprofile.health;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+import org.eclipse.microprofile.health.Readiness;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+@Readiness
+@ApplicationScoped
+public class DatabaseConnectionHealthCheck implements HealthCheck {
+
+    @Inject
+    @ConfigProperty(name = "database.up", defaultValue = "false")
+    private boolean databaseUp;
+
+    @Override
+    public HealthCheckResponse call() {
+
+        HealthCheckResponseBuilder responseBuilder = HealthCheckResponse.named("Database connection health check");
+
+        try {
+            simulateDatabaseConnectionVerification();
+            responseBuilder.up();
+        } catch (IllegalStateException e) {
+            // cannot access the database
+            responseBuilder.down();
+        }
+
+        return responseBuilder.build();
+    }
+
+    private void simulateDatabaseConnectionVerification() {
+        if (!databaseUp) {
+            throw new IllegalStateException("Cannot contact database");
+        }
+    }
+}
+----
+
+Redeploy your application:
+
+[source,options="nowrap"]
+----
+$ mvn clean package wildfly:deploy
+----
+
+If you now rerun the readiness health check (at `http://localhost:9990/health/ready`)
+the overall `status` should be DOWN. You can also check the liveness check at
+`http://localhost:9990/health/live` which will return the overall `status` UP because
+it isn't influenced by the readiness checks.
+
+As we shouldn't leave this application with a readiness check in a DOWN state and
```

Comment:
section about quarkus does not apply, right?

Reply:
agh :) thanks a lot!